refinery/data_set_manager/single_file_column_parser.py

```diff
@@ -181,12 +184,10 @@ def run(self):
         # import in file as "pre-isa" file
         logger.info("trying to add pre-isa archive file %s",
                     self.metadata_file.name)
-        # FIXME: this will not create a FileStoreItem if self.metadata_file
-        # does not exist on disk (e.g., a file object like TemporaryFile)
         file_store_item = \
-            FileStoreItem.objects.create(source=self.metadata_file.name)
+            FileStoreItem.objects.create(source=self.metadata_file_path)
```

Comment:
The only change that is needed:
```python
file_store_item = FileStoreItem.objects.create(source=metadata_file.name)
file_store_item.datafile.save(os.path.basename(metadata_file.name),
                              django.core.files.File(metadata_file))
```
https://docs.djangoproject.com/en/1.8/ref/models/fields/#django.db.models.fields.files.FieldFile.save